### PR TITLE
ISPN-12372 docs for storageclass

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12372

### CODE REVIEW

File: documentation/asciidoc/topics/ref_container_storage.adoc [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+for {datagridservice} nodes but not {cacheservice} nodes.
 
 [source,options="nowrap",subs=attributes+]
 ----
 include::yaml_annotated/cr_container_storage.yaml[]
 ----
 
 <1> Configures the storage size for {datagridservice} nodes.
+<2> Defines whether storage is ephemeral or not. Set the value to `true` to use ephemeral storage, which means all data in storage is deleted when clusters shut down or restart. The default value is `false`.
+<3> Specifies the name of a StorageClass object to use for the persistent volume claim.
```
Comment (oraNod) [inline documentation/asciidoc/topics/ref_container_storage.adoc:14]: @dmvolod Is there any default for the StorageClass? is it standard? what happens if the user omits this or leaves a null value?

File: documentation/asciidoc/topics/ref_container_storage.adoc [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+{cacheservice} and {datagridservice} nodes. You can configure storage resources
+for {datagridservice} nodes but not {cacheservice} nodes.
 
 [source,options="nowrap",subs=attributes+]
 ----
 include::yaml_annotated/cr_container_storage.yaml[]
 ----
 
 <1> Configures the storage size for {datagridservice} nodes.
+<2> Defines whether storage is ephemeral or not. Set the value to `true` to use ephemeral storage, which means all data in storage is deleted when clusters shut down or restart. The default value is `false`.
```
Comment (oraNod) [inline documentation/asciidoc/topics/ref_container_storage.adoc:13]: @dmvolod default is false, yeah? so in previous versions data in the PVC is permanent - hence "persistent volume claim"?

**Review by oraNod — COMMENTED** [review]
@dmvolod In docs we also state "Persistent volume claims use the ReadWriteOnce (RWO) access mode." - Is this still true by default? If users specify a different StorageClass can they use a different access mode?

File: documentation/asciidoc/topics/ref_container_storage.adoc (hunk 2 above)
Comment (dmvolod) [reply documentation/asciidoc/topics/ref_container_storage.adoc:13]: Yes, true. Infinispan by default will always use a permanent PVC

File: documentation/asciidoc/topics/ref_container_storage.adoc (hunk 1 above)
Comment (dmvolod) [reply documentation/asciidoc/topics/ref_container_storage.adoc:14]: @oraNod the default Storage Class name depends of the platform where k8s/ocp/okd is deploying.
https://kubernetes.io/docs/concepts/storage/storage-classes
If user set empty or not exists Storage Class name operator will print an error that it's not exists.
If Storage Class name is not provided, the PVC will use storage class which has an annotation storageclass.kubernetes.io/is-default-class set to true.
https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/

File: documentation/asciidoc/topics/ref_datagridservice_resources.adoc [hunk 3]
```diff
@@ -11,9 +11,14 @@ spec:
   replicas: 6 <2>
   service:
     type: DataGrid <3>
-    container:
-      storage: 2Gi <4>
-    sites: <5>
+    container: <4>
+      extraJvmOpts: "-XX:NativeMemoryTracking=summary"
```
Comment (dmvolod) [inline documentation/asciidoc/topics/ref_datagridservice_resources.adoc:15]: @oraNod looks like extraJvmOpts, cpu and memory and storage parameters are from different container levels

**Comment by dmvolod** [comment]
> In docs we also state "Persistent volume claims use the ReadWriteOnce (RWO) access mode." - Is this still true by default? If users specify a different StorageClass can they use a different access mode?

yes, true. We don't need another access mode rather than ReadWriteOnce (RWO) as it hard-coded in the operator.

File: documentation/asciidoc/topics/ref_datagridservice_resources.adoc (hunk 3 above)
Comment (oraNod) [reply documentation/asciidoc/topics/ref_datagridservice_resources.adoc:15]: @dmvolod thanks, for some reason I thought `spec.container` and `spec.service.container` were dupes. I guess that's what happens when you have too many meetings in one day. fixed.

Comment (dmvolod) [reply documentation/asciidoc/topics/ref_datagridservice_resources.adoc:15]: No, but I'm also confusing `container's` sometimes. But not sure that's possible to change this structure dramatically...

**Review by dmvolod — APPROVED** [review]

**Review by ryanemerson — APPROVED** [review]

**Pull request merged**